# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaStreamGraph.h
# Commit: c54cb3c10992
# Full Hash: c54cb3c109922bc15fb6d0ca67e5f2a05980c1a5
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-07-31 15:46:43
# Regressor Bug: 1493613
# File Overlap Count: 2
# Description:
#   Bug 1493613 - Move MediaStream control from DOMMediaStream to MediaStreamTrack. r=padenot
#   
#   This is inherently large, because modifying these bits of DOMMediaStream and
#   MediaStreamTrack affects all consumers and producers of all DOMMediaStreams and
#   MediaStreamTracks.
# ==============================================================================

diff -r 0b03dd9d20ac -r c54cb3c10992 dom/media/MediaStreamGraph.h
--- a/dom/media/MediaStreamGraph.h	Wed Jul 31 09:35:06 2019 +0000
+++ b/dom/media/MediaStreamGraph.h	Wed Jul 31 07:58:17 2019 +0000
@@ -28,7 +28,8 @@
 
 namespace mozilla {
 class AsyncLogger;
-};
+class AudioCaptureStream;
+};  // namespace mozilla
 
 extern mozilla::AsyncLogger gMSGTraceLogger;
 
@@ -365,17 +366,8 @@
   void RunAfterPendingUpdates(already_AddRefed<nsIRunnable> aRunnable);
 
   // Signal that the client is done with this MediaStream. It will be deleted
-  // later. Do not mix usage of Destroy() with RegisterUser()/UnregisterUser().
-  // That will cause the MediaStream to be destroyed twice, which will cause
-  // some assertions to fail.
+  // later.
   virtual void Destroy();
-  // Signal that a client is using this MediaStream. Useful to not have to
-  // explicitly manage ownership (responsibility to Destroy()) when there are
-  // multiple clients using a MediaStream.
-  void RegisterUser();
-  // Signal that a client no longer needs this MediaStream. When the number of
-  // clients using this MediaStream reaches 0, it will be destroyed.
-  void UnregisterUser();
 
   // Returns the main-thread's view of how much data has been processed by
   // this stream.
@@ -618,7 +610,6 @@
   bool mMainThreadFinished;
   bool mFinishedNotificationSent;
   bool mMainThreadDestroyed;
-  int mNrOfMainThreadUsers;
 
   // Our media stream graph.  null if destroyed on the graph thread.
   MediaStreamGraphImpl* mGraph;
@@ -1226,7 +1217,7 @@
   /**
    * Create a stream that will mix all its audio input.
    */
-  ProcessedMediaStream* CreateAudioCaptureStream(TrackID aTrackId);
+  AudioCaptureStream* CreateAudioCaptureStream(TrackID aTrackId);
 
   /**
    * Add a new stream to the graph.  Main thread.